### PR TITLE
Set groupid when setting provider traits

### PR DESCRIPTION
**Pull request by MarkWestMS**
This PR addresses an issue where the `provider_group_guid` is not being added to the provider traits during registration.

### Causes

Both `register_provider_metadata` and `set_provider_traits` are being called, each with a different trait buffer.  These are both backed by `EventSetInformation`, which can only be called once because subsequent calls will fail ([documentation](https://learn.microsoft.com/en-us/windows/win32/etw/provider-traits)).  This failure is normally not surfaced because the error is ignored in `create_register_provider_traits`.

The trait buffer passed to set the provider group is not writing the bytes as fields as expected by `EventSetInformation`.

The trait buffer is being aligned, but it appears it is not needed and will cause `EventSetInformation` to fail depending on provider name length.

### Changes

Remove the `register_provider_metadata` call so that `set_provider_traits` will succeed.

Remove the trait buffer alignment calls and write the GUID as fields.

### CODE REVIEW

File: win_etw_macros/src/lib.rs [hunk 1]
```diff
@@ -668,32 +667,35 @@ fn create_register_provider_traits(
...
 
     if let Some(provider_group_guid) = provider_group_guid {
         // Add trait for provider guid
         let provider_guid_trait_offset = traits_bytes.len();
         traits_bytes.push(0); // reserve space for TraitSize (u16)
         traits_bytes.push(0);
         traits_bytes.push(ETW_PROVIDER_TRAIT_TYPE_GROUP);
-        traits_bytes.extend_from_slice(provider_group_guid.as_bytes());
+
+        let (data1, data2, data3, data4) = provider_group_guid.to_fields_le();
```
Comment (sivadeilra) [inline win_etw_macros/src/lib.rs:684]: Ahh, so I had the repr of the UUID wrong.

Can this just be `traits_bytes.extend_from_slice(&provider_group_guid.to_bytes_le());`?

If not, can we move the guts of this to a separate method, like:

```
fn uuid_to_bytes_le(g: &Guid) -> [u8; 16] {
    let (d1, d2, d3, d4) = g.to_fields_le();
    let d1 = d1.to_be_bytes();
    let d2 = d2.to_be_bytes();
    let d3 = d3.to_be_bytes();
    // d4 is already bytes
    [
        d1[0], d1[1], d1[2], d1[3],
        d2[0], d2[1],
        d3[0], d3[1],
        d4[0], d4[1], d4[2], d4[3], d4[4], d4[5], d4[6], d4[7],
    ]
  }
}

...
   traits_bytes.push(uuid_to_le_bytes(&provider_group_guid));
```

File: win_etw_macros/src/lib.rs [hunk 2]
```diff
@@ -612,7 +612,6 @@ fn trace_logging_events_core(attr: TokenStream, item_tokens: TokenStream) -> Tok
                     Ok(mut provider) => {
                         #[cfg(target_os = "windows")]
                         {
-                            let _ = provider.register_provider_metadata(&#provider_metadata_ident);
```
Comment (sivadeilra) [inline win_etw_macros/src/lib.rs:615]: Why is this removed?

File: win_etw_macros/src/lib.rs [hunk 3]
```diff
@@ -668,32 +667,35 @@ fn create_register_provider_traits(
...
+        traits_bytes.push(((data2 & 0x00ff) >> 0) as u8);
+        traits_bytes.push(((data3 & 0xff00) >> 8) as u8);
+        traits_bytes.push(((data3 & 0x00ff) >> 0) as u8);
+        traits_bytes.extend_from_slice(data4);
+
         let provider_guid_trait_len = traits_bytes.len() - provider_guid_trait_offset;
         // Set TraitSize (u16)
         traits_bytes[provider_guid_trait_offset] = provider_guid_trait_len as u8;
         traits_bytes[provider_guid_trait_offset + 1] = (provider_guid_trait_len >> 8) as u8;
-        align2(&mut traits_bytes);
```
Comment (sivadeilra) [inline win_etw_macros/src/lib.rs:696]: Might be good to convert `align2(...)` to `assert!(is_aligned_2(&traits_bytes));`.

**Review by sivadeilra — CHANGES_REQUESTED** [review]
Thanks for the fix!  Happy to get this working.

File: win_etw_macros/src/lib.rs (hunk 2 above)
Comment (sivadeilra) [reply win_etw_macros/src/lib.rs:615]: Oh, you already preempted this question!  Sorry.

File: win_etw_macros/src/lib.rs (hunk 3 above)
Comment (idigdoug) [reply win_etw_macros/src/lib.rs:696]: Traits are packed tightly and are not aligned.

**Review by sivadeilra — APPROVED** [review]
